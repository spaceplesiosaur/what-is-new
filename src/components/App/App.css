* {
  box-sizing: border-box;
  box-sizing: inherit;
  margin: 0;
  padding: 0;
}

.app {
  display: flex;
  background: #030D4D;
  height: 100vh;
  text-align: center;
  width: 100%;
}

.main {
  display: flex;
  flex-direction: column;
  width: 100%;
}
