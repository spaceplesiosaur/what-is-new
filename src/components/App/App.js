import React, { Component } from 'react';
// import local from '../../data/local';
// import entertainment from '../../data/entertainment';
// import health from '../../data/health';
// import science from '../../data/science';
// import technology from '../../data/technology';
import './App.css';
import SearchForm from '../SearchForm/SearchForm.js';
import Menu from '../Menu/Menu.js';
import NewsContainer from '../NewsContainer/NewsContainer.js'

class App extends Component {
  constructor() {
    super();
    this.state = {
  
      topic: [],

      menu:
        [
          {name: 'Entertainment', image: '/favicon.ico', id: []},
          {name: 'Health', image: '/favicon.ico', id: []},
          {name: 'Local News', image: '/favicon.ico', id: []},
          {name: 'Science', image: '/favicon.ico', id: []},
          {name: 'Technology', image: '/favicon.ico', id: []}
        ]

    }
  }
  componentDidMount = () => {
   fetch('https://whats-new-api.herokuapp.com/api/v1/news')
     .then(response => response.json())
     .then(data => this.setState({ topic: data.local, menu: this.state.menu.map((item, i) => {
       return {name: item.name, image: item.image, id: data[Object.keys(data)[i]]}
     })
   }));
 }
  setTopic = (topicId) => {
    this.setState({topic: topicId})
  }

  filterArticles = (search) => {
     const filteredTopic = [...this.state.topic].filter(article => {
      return article.headline.includes(search) ||
      article.description.includes(search);
    })
    this.setState({topic: filteredTopic})
  }

  render () {
    return (
      <div className="app">
        <Menu
        menuItems={this.state.menu}
        setTopic={this.setTopic}
        />
        <SearchForm
        filterArticles={this.filterArticles}
        />
        <NewsContainer
        articleList={this.state.topic}
        />
      </div>
    );
  }
}


export default App;
