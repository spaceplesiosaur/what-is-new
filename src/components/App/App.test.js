import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should render app', () => {
  const wrapper = shallow(<App
  />);
  expect(wrapper).toMatchSnapshot();
});

// it('should set a topic based on menu selection', () => {
//   const app = new App();
//   app.setTopic(entertainment)
//
// })
