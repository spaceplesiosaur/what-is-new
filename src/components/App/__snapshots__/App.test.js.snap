// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render app 1`] = `
<div
  className="app"
>
  <Menu
    menuItems={
      Array [
        Object {
          "id": Array [],
          "image": "/film-solid.png",
          "name": "Entertainment",
        },
        Object {
          "id": Array [],
          "image": "/heart-solid.png",
          "name": "Health",
        },
        Object {
          "id": Array [],
          "image": "/map-marker-alt-solid.png",
          "name": "Local News",
        },
        Object {
          "id": Array [],
          "image": "/atom-solid.png",
          "name": "Science",
        },
        Object {
          "id": Array [],
          "image": "/cogs-solid.png",
          "name": "Technology",
        },
      ]
    }
    setTopic={[Function]}
  />
  <section
    className="main"
  >
    <SearchForm
      filterArticles={[Function]}
    />
    <NewsContainer
      articleList={Array []}
    />
  </section>
</div>
`;
